addons/web/static/tests/views/kanban/kanban_view_tests.js

```diff
@@ -5021,6 +5021,85 @@ QUnit.module("Views", (hooks) => {
         assert.verifySteps(["resequence"]);
     });
 
+    QUnit.test("user without permission cannot drag and drop a column thus sequence remains unchanged on drag and drop attempt", async (assert) => {
+        let hasPermission = false; // Simulate user without permission
+
+        await makeView({
+            type: "kanban",
+            resModel: "partner",
+            serverData,
+            arch:
+                '<kanban on_create="quick_create">' +
+                '<field name="product_id"/>' +
+                '<templates><t t-name="kanban-box">' +
+                '<div class="oe_kanban_global_click"><field name="foo"/>' +
+                '<t t-if="widget.editable"><span class="thisiseditable">edit</span></t>' +
+                "</div>" +
+                "</t></templates>" +
+                "</kanban>",
+            groupBy: ["product_id"],
+            async mockRPC(route) {
+                if (route === "/web/dataset/resequence") {
+                    if (!hasPermission) {
+                        return Promise.reject();    // Simulate no permission
+                    }
+                }
+            },
+        });
+
+        let groups = target.querySelectorAll(".o_column_title");
+        await dragAndDrop(
+            groups[0],
+            groups[1]
+        ).then(() => {
+            groups = target.querySelectorAll(".o_column_title");
+            assert.strictEqual(groups[0].textContent, "hello", 'Do not let the user arrange the columns without permission');
+            assert.strictEqual(groups[1].textContent, "xmo", 'Do not let the user arrange the columns without permission');
+        });
+    });
+
+    QUnit.test("user without permission cannot drag and drop a record thus sequence remains unchanged on drag and drop attempt", async (assert) => {
+        let hasPermission = false; // Simulate user without permission
+
+        await makeView({
+            type: "kanban",
+            resModel: "partner",
+            serverData,
+            arch:
+                '<kanban on_create="quick_create">' +
+                '<field name="product_id"/>' +
+                '<templates><t t-name="kanban-box">' +
+                '<div class="oe_kanban_global_click"><field name="foo"/>' +
+                '<t t-if="widget.editable"><span class="thisiseditable">edit</span></t>' +
+                "</div>" +
+                "</t></templates>" +
+                "</kanban>",
```

Comment:
you don't need a such complex arch
```suggestion
            arch: `
                <kanban>
                    <templates>
                        <t t-name="kanban-box">
                            <div class="oe_kanban_global_click">
                                <field name="foo"/>
                            </div>
                        </t>
                    </templates>
                </kanban>`,
```

Same for the other test